nntrainer/cl_context.cpp

```diff
@@ -32,25 +32,33 @@ std::once_flag global_cl_context_init_flag;
 
 static void add_default_object(ClContext &cc) {
 
-  cc.registerFactory(nntrainer::createLayer<FullyConnectedLayerCl>,
-                     FullyConnectedLayerCl::type,
-                     ml::train::LayerType::LAYER_FC);
+  if (FullyConnectedLayerCl::registerClKernels()) {
```

Comment:
currently, registerClKernels() always return true. are there going to be cases where it returns false?

Reply:
Yeap. In current version, I just throw exception. But, I think it would be better to update the code to return false and register it if it fails to register (to allow nntrainer to retry it later).

Reply:
I updated the code to return true only when all the kernels are registered successfully.
Please refer to: 
https://github.com/nnstreamer/nntrainer/blob/6dc250f57117375d5d55a981976f59cdf4e8540c/nntrainer/layers/cl_layers/concat_cl.cpp#L302